Swiftz/NonEmptyList.swift

```diff
@@ -50,12 +50,12 @@ extension NonEmptyList : ArrayLiteralConvertible {
 	public init(arrayLiteral s: Element...) {
 		var xs : [A] = []
 		var g = s.generate()
-		let h: A? = g.next()
+		let h : A? = g.next()
 		while let x : A = g.next() {
 			xs.append(x)
 		}
 		var l = List<A>()
-		for x in Array(xs.reverse()) {
+		for x in NonEmptyList(List(fromArray: xs.reverse()))! {
```

Comment:
While I'm thinking about it: should we make this `init!(arrayLiteral:)` instead of trapping unconditionally?  The use of implicitly-unwrapped optionals should mean `swiftc` is fine with the overriding of an `init` with an `init!`.